instaloader/instaloadercontext.py

```diff
@@ -256,18 +255,24 @@ def login(self, user, passwd):
         # Override default timeout behavior.
         # Need to silence mypy bug for this. See: https://github.com/python/mypy/issues/2427
         session.request = partial(session.request, timeout=self.request_timeout) # type: ignore
-        csrf_json = self.get_json('accounts/login/', {}, session=session)
-        csrf_token = csrf_json['config']['csrf_token']
+
+        self.do_sleep()
```

Comment:
At this point `do_sleep()` seems unnecessary to me, no request has been sent here, so the function waits at the start before doing anything.

Reply:
In the old code, `get_json` (which has a `do_sleep()` at the start) is called at the start of login. Since my changes made `get_json` not used during login, I moved the `do_sleep()` to before the request. Feel free to remove it if you think it's unnecessary.

Reply:
Oh indeed, thanks for that careful observation. Probably that was to defer the request rather than the processing of data it responds.